New: Blog post announcing v7.0.0 features

nzakas opened the pull request:
This is a blog post previewing the significant features and changes coming in v7.0.0. I did my best to explain the rationale behind some of the bigger changes, but do let me know if I didn't get it right.

Note that I only included what I thought to be significant changes, so please feel free to let me know if I missed anything. I didn't think changes to rules were important enough to merit inclusion (those will be better explained in the release notes).

platinumazure commented:
This post mentions releases happening every two weeks, but (as of when I was on the team) we were releasing every four weeks. Is ESLint releasing every two weeks again? (Kai was hired on as dedicated maintainer after the decision to go to four weeks, so the landscape definitely changed after we made that decision.)

kaicataldo commented:
@platinumazure We're not releasing every two weeks, though I'm happy to be the point person to do so.

kaicataldo commented on _posts/2020-02-20-whats-coming-in-eslint-7.0.0.md:
```diff
@@ -0,0 +1,105 @@
+
+### File extensions in config files
+
+ESLint v7.0.0 now reads the glob patterns in the `overrides` section of your config to determine while files to lint. Prior to v7.0.0, you would need to provide the `--ext` command line flag to specify the additional files to lint. You can read more in the [RFC](https://github.com/eslint/rfcs/pull/20).
```
```suggestion
ESLint v7.0.0 now reads the glob patterns in the `overrides` section of your config to determine which files to lint. Prior to v7.0.0, you would need to provide the `--ext` command line flag to specify the additional files to lint. You can read more in the [RFC](https://github.com/eslint/rfcs/pull/20).
```

kaicataldo commented on _posts/2020-02-20-whats-coming-in-eslint-7.0.0.md:
```diff
@@ -0,0 +1,105 @@
+
+### Updated default ignore patterns
+
+The default ignore patterns for ESLint will be changed to:
```
Can we make it clear that these are additive changes? Maye something like:
```suggestion
In addition to the current defaults, the default ignore patterns for ESLint will be changed to:
```

kaicataldo commented:
Thanks for writing this up! Two small comments, but otherwise LGTM.

nzakas commented:
Thanks. Apparently I can’t merge changes from my phone so I’ll have to do that later.

I didn’t realize we had changed to every four weeks for releases. Is there any background to that decision you can point me to? FWIW, the readme still says every two weeks.

mysticatea commented on _posts/2020-02-20-whats-coming-in-eslint-7.0.0.md:
```diff
@@ -0,0 +1,105 @@
+
+### Deprecating `~/.eslintrc`
+
+Config files in your home directory are deprecated in v7.0.0. This feature has long been a source of confusion for ESLint users, especially because ESLint only looks for a config file in the home directory if no other config file is found. The configuration system in ESLint has changed a lot since home directory config files were introduced, and with the additional flexibility we feel there are p ...
```
This deprecation happened in 6.7.0. We add runtime warnings in 7.0.0 for a few breaking cases on removal.

nzakas replied:
Ooh thanks, I'll update that.

ilyavolodin commented:
Switch to once-per-4 weeks release happened as part of TSC discussion a few month ago. The reasons for it were that we couldn't find people to cover releases every 2 week. Suggestions were to either switch back to requiring one person per release, or to switch to release every 4 weeks. We missed a bunch of scheduled releases in December because only one person was available to cover release.

nzakas commented:
@ilyavolodin thanks for the details.

nzakas commented:
Okay, I've addressed all of the comments. Please take a look and let me know if anything else should be updated before publishing. (I'd like to publish today, so please take a look as soon as you can.)

kaicataldo commented:
Thanks for writing this up!

mysticatea commented on _posts/2020-02-20-whats-coming-in-eslint-7.0.0.md:
```diff
@@ -58,16 +58,19 @@ Directive comment descriptions allow you to leave a note to yourself and other d
 
+### Plugins loaded from config file directory
+
+ESLint has long had a problem loading plugins from shared configs due to the way plugin loading was handled. This problem required using peer dependencies and a post-install script to allow shared configs to reference plugins. In v7.0.0, plugins will be loaded relative to the configs that reference them, allowing shared configs to include plugins as dependencies directly. You can read more in the ...
```
> allowing shared configs to include plugins as dependencies directly.

No.
RFC47 doesn't intend to allow shared configs to include plugins as dependencies directly. See detail: https://github.com/eslint/rfcs/blob/master/designs/2019-plugin-loading-improvement/README.md#the-plugin-uniqueness-with-shareable-configs

nzakas replied:
Oh thanks. That RFC was a bit hard to understand for me, so I missed this part.
